underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -331,7 +333,8 @@ protected ObjectListingChunk getObjectListingChunk(String key, boolean recursive
 
     @Override
     public String[] getCommonPrefixes() {
-      // TODO(adit): infer directories
+      // When a delimiter is used, the Swift backend infers pseudo-directories and returns the
+      // directory names (including a trailing '/') as part of the object names returned.
```

Comment:
Not something that needs to go into this PR. Do we want to reconcile this behavior, since `getCommonPrefixes` is part of the API of `ObjectListingChunk`? Also, do we want to have unit tests to enforce the API of this interface?